dcm/src/main/java/org/dcm/viewupdater/DDlogUpdater.java

```diff
@@ -0,0 +1,121 @@
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class DDlogUpdater {
+    private final DDlogAPI API;
+    private final Map<String, Integer> tableIDMap;
+    private final Map<String, IRTable> irTables;
+    private final Consumer<DDlogCommand<DDlogRecord>> consumer;
+
+    static final String INTEGER_TYPE = "java.lang.Integer";
+    static final String STRING_TYPE = "java.lang.String";
+    static final String BOOLEAN_TYPE = "java.lang.Boolean";
+    static final String LONG_TYPE = "java.lang.Long";
+
+    public DDlogUpdater(final Consumer<DDlogCommand<DDlogRecord>> consumer, final Map<String, IRTable> irTables) {
+        final int ddlogWorkerThreads = 1;
+        final boolean storeDataInDDlogBackgroundProgram = false;
+        try {
+            API = new DDlogAPI(ddlogWorkerThreads, null, storeDataInDDlogBackgroundProgram);
+        } catch (final DDlogException e) {
+            throw new RuntimeException(e);
+        }
+
+        this.consumer = consumer;
+        this.tableIDMap = new HashMap<>();
+        this.irTables = irTables;
+    }
+
+    private DDlogRecord toDDlogRecord(final String tableName, final Object[] args) {
+        final DDlogRecord[] recordsArray = new DDlogRecord[args.length];
+        final IRTable irTable = irTables.get(tableName);
+        final Table<? extends Record> table = irTable.getTable();
+
+        int counter = 0;
+        for (final Field<?> field : table.fields()) {
+            final Class<?> cls = field.getType();
+            try {
+                switch (cls.getName()) {
+                    case BOOLEAN_TYPE:
+                        recordsArray[counter] = new DDlogRecord((Boolean) args[counter]);
+                        break;
+                    case INTEGER_TYPE:
+                        recordsArray[counter] = new DDlogRecord((Integer) args[counter]);
+                        break;
+                    case LONG_TYPE:
+                        recordsArray[counter] = new DDlogRecord((Long) args[counter]);
+                        break;
+                    case STRING_TYPE:
+                        recordsArray[counter] = new DDlogRecord(args[counter].toString().trim());
+                        break;
+                    default:
+                        throw new RuntimeException("Unexpected datatype: " + cls.getName());
+                }
+            } catch (final DDlogException e) {
+                throw new RuntimeException(e);
+            }
+            counter = counter + 1;
+        }
+        DDlogRecord record = null;
+        try {
+            record = DDlogRecord.makeStruct(tableName, recordsArray);
+        } catch (final DDlogException e) {
+            throw new RuntimeException(e);
+        }
+        return record;
+    }
+
+    public void sendUpdatesToDDlog(final Map<String, List<Object[]>> commands) {
+        final int counter = commands.values().stream().mapToInt(List::size).sum();
```

Comment:
What exactly is going on with this counter? It is not being used a counter, but contains the sum of all record sizes internally?

I think you want to call it numCommands?